### DISCUSSION TITLE
Check-in

### DISCUSSION THREAD

**Discussion by terminator356** [muse-sequencer/muse#1181] [Team Posts]
Been working several days on that latency code crash issue.
Making some progress.
But gosh, my latency code and logic are hard to remember.
Especially when some of it is clearly causing the crash.
"What was the logic here, again? it's wrong so what's the right way then...?"

**Comment by spamatica** [comment]
Hehe, preach! I feel like that often :P
Anyways, it's rather stable for me.
Have this feeling it sometimes gets the latency compensating wrong though. Hard to quantify, could be in my head.. If there is something I think it has to do with xruns.

At the moment not much happening here with coding... started my intermission year at school though😁

**Comment by terminator356** [comment]
Decrease the buffer size in your head :-)

I wish I could automate the latency testing process somehow but physical cables are required
 and so many variations of test setups.
Keep in mind I still have to do Aux tracks, which I know that you like to use.

What is an intermission year, could this be a typo?

**Comment by spamatica** [comment]
Hehe, latency in my head is sadly constant :P 

True with latency testing though, hmm. Is it possible to somehow measure the latency in realtime? 
... guessing not, but now that you mention it, i did test latency before by examining the recorded audio. 
I should be able to redo those kinds of tests and punching MusE in the gut once in a while to see if it at some point changes.

Heh, me trying to use more english words ;) I think I mentioned that I was going back to school. I'm having a year off work (...sooo intermission from work ;) ) and attending a pretty laid back art school. So far it's been very good for me.

**Comment by terminator356** [comment]
> Is it possible to somehow measure the latency in realtime?

Yes but not in the background while the software is actually being used.
We run some tests, some involving physical loop-back cables, that send
 special signals whose latency can be measured. (See how jack_iodelay does this.)

But there are actually more tests than those.
For example audio, midi, metronome, and synth audio+midi latency.

A series of MusE projects would help with the tests.
I have some and had hoped to publish them but they are all mixed up and have
 multiple versions (as I fixed bugs).
In fact there were so many uniquely different specific tests for every little thing.
It became hard to decide which ones were important enough to publish vs. which ones
 were long tedious tests for strange behaviour bugs, corrections etc.
